src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -1421,92 +1424,152 @@ bool OutputStateMachineEngine::s_ParseColorSpec(const std::wstring_view string,
+    std::array<float, 3> colorValues = { 0 };
+    std::array<unsigned int, 3> parameterValues = { 0 };
     bool success = false;
-    // We can have anywhere between [11,15] characters
-    // 9 "rgb:h/h/h"
-    // 12 "rgb:hh/hh/hh"
-    // Any fewer cannot be valid, and any more will be too many.
-    // Return early in this case.
-    //      We'll still have to bounds check when parsing the hh/hh/hh values
-    if (string.size() < 9 || string.size() > 12)
-    {
-        return false;
-    }
+    const auto stringSize = string.size();
 
-    // Now we look for "rgb:"
+    // First we look for "rgb:"
     // Other colorspaces are theoretically possible, but we don't support them.
-
     auto curr = string.cbegin();
     if ((*curr++ == L'r') &&
         (*curr++ == L'g') &&
         (*curr++ == L'b') &&
         (*curr++ == L':'))
     {
+        // We can have one of the following formats:
+        // 9 "rgb:h/h/h"
+        // 12 "rgb:hh/hh/hh"
+        // 15 "rgb:hhh/hhh/hhh"
+        // 18 "rgb:hhhh/hhhh/hhhh"
+        // Any other cases will be invalid.
+        if (!(stringSize == 9 || stringSize == 12 || stringSize == 15 || stringSize == 18))
+        {
+            return false;
+        }
+
+        foundRGB = true;
+        rgbHexDigitCount = ((stringSize - 3) / 3) - 1;
+    }
+
+    // Try the sharp sign format.
+    curr--;
+    if (*curr++ == L'#')
+    {
+        // We can have one of the following formats:
+        // 4 "#hhh"
+        // 7 "#hhhhhh"
+        // 10 "#hhhhhhhhh"
+        // 13 "#hhhhhhhhhhhh"
+        // Any other cases will be invalid.
+        if (!(stringSize == 4 || stringSize == 7 || stringSize == 10 || stringSize == 13))
+        {
+            return false;
+        }
+
+        isSharpSignFormat = true;
         foundRGB = true;
+        rgbHexDigitCount = (stringSize - 1) / 3;
     }
 
     if (foundRGB)
     {
-        // Colorspecs are up to hh/hh/hh, for 1-2 h's
+        // Colorspecs are up to hhhh/hhhh/hhhh, for 1-4 h's
         for (size_t component = 0; component < 3; component++)
         {
             bool foundColor = false;
-            auto& value = til::at(colorValues, component);
-            for (size_t i = 0; i < 3; i++)
+            auto& parameterValue = til::at(parameterValues, component);
+            for (size_t i = 0; i < rgbHexDigitCount; i++)
             {
                 const wchar_t wch = *curr++;
 
                 if (_isHexNumber(wch))
                 {
-                    value *= 16;
+                    parameterValue *= 16;
                     unsigned int intVal = 0;
                     if (s_HexToUint(wch, intVal))
                     {
-                        value += intVal;
+                        parameterValue += intVal;
                     }
                     else
                     {
                         // Encountered something weird oh no
                         foundColor = false;
                         break;
                     }

```

Comment:
As far as I can tell, this serves no purpose. The `foundColor` flag starts off as false and is never set to true anywhere in this loop. And then the code that follows is going to immediately overwrite the value anyway, so it doesn't make any difference. It looks like most error cases aren't actually caught. For example, this seems to be a perfectly acceptable representation for the color black: `rgb:this/ is /good`